Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/TierDeleted/TierDeletedIntegrationEventHandler.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Microsoft.Extensions.Logging;
+
+namespace Backbone.Modules.Quotas.Application.IntegrationEvents.Incoming.TierDeleted;
+public class TierDeletedIntegrationEventHandler : IIntegrationEventHandler<TierDeletedIntegrationEvent>
+{
+    private readonly ITiersRepository _tiersRepository;
+    private readonly ILogger<TierDeletedIntegrationEventHandler> _logger;
+
+    public TierDeletedIntegrationEventHandler(ILogger<TierDeletedIntegrationEventHandler> logger, ITiersRepository tiersRepository)
+    {
+        _tiersRepository = tiersRepository;
+        _logger = logger;
+    }
+
+    public async Task Handle(TierDeletedIntegrationEvent integrationEvent)
+    {
+        var tier = await _tiersRepository.Find(integrationEvent.Id, CancellationToken.None);
+
+        await _tiersRepository.Remove(tier);
```

Comment:
Now that I see this, I think that since we don't perform any checks on the retrieved Tier, I think we can as well just have a `RemoveById` method on the repository. This would save us one database roundtrip

Reply:
What EF method should I use in this scenario? I can only find `Remove` and `RemoveRange`.

Reply:
How about https://stackoverflow.com/a/75185453/4046585